packages/vm/tests/BlockchainTestsRunner.js

```diff
@@ -104,6 +102,15 @@ module.exports = async function runBlockchainTest(options, testData, t) {
     const expectException = raw[paramFork] ? raw[paramFork] : raw[paramAll1] || raw[paramAll2] || raw.blockHeader == undefined
 
     try {
+      // check if we should update common.
+      if (common.isNextHardforkBlock(currentBlock)) {
+        const activeHardforks = common.activeHardforks(currentBlock)
+        const hardforkName = activeHardforks[activeHardforks.length - 1].name
+        common.setHardfork(hardforkName)
+        // create a new VM (need access to new opcodes)
+        vm._updateOpcodes()
+      }
+
```

Comment:
This can be simplified with [Common.setHardforkByBlockNumber()](https://github.com/ethereumjs/ethereumjs-vm/blob/master/packages/common/src/index.ts#L163)

Reply:
Yeah there is some weird stuff going on over here, because if I re-instantiate a VM after I've updated Common (assuming that is what you mean here) then suddenly a lot more tests start failing. This used to be the way how most blockchain/state tests passed. 🤔

Reply:
It's worth investigating why this happens though as it should definitely work if we re-instantiate the VM with a correctly-set Common 👍